edweiss-app/app/(app)/lectures/slides/index.tsx

```diff
@@ -48,62 +47,47 @@ const LectureScreen: ApplicationRoute = () => {
         }
     });
 
-
     // Hooks
     const [numPages, setNumPages] = useState<number>(0);        // Total number of pages in the PDF
     const [page, setPage] = useState<number>(0);                // Current page, starting from 1
     const [uri, setUri] = useState<string>('');                 // Url state
-    //const [isLandscape, setIsLandscape] = useState(false);
-    const [isFullscreen, setIsFullscreen] = useState(false);    // FullScreen display of pdf toggle
+    // UI display setting's hooks
+    const [isLandscape, setIsLandscape] = useState<boolean>(false);       // Landscape display boolean for different UI
+    const [isFullscreen, setIsFullscreen] = useState<boolean>(false);    // FullScreen display of pdf toggle
+    const [widthPercent, setWidthPercent] = useState<string[]>([]);
+    const [heightPercent, setHeightPercent] = useState<string[]>([]);
 
     const [lectureDoc] = usePrefetchedDynamicDoc(CollectionOf<Lecture>(`courses/${courseName}/lectures`), lectureId, undefined);
 
     // Load images when the component mounts
     useEffect(() => {
         if (lectureDoc) {
             getUri();
-            setLandscape();
         }
+    }, [lectureDoc]);
 
-        /* todo for a next sprint, detection of rotation change from landscape to portrait and different UI for each mode
-
-        // Orientation change listener
-        const subscription = ScreenOrientation.addOrientationChangeListener(({ orientationInfo }) => {
-            const { orientation } = orientationInfo;
-            // If the orientation is portrait (either portrait up or down), trigger setPortrait
-            if (
-                orientation === ScreenOrientation.Orientation.PORTRAIT_UP ||
-                orientation === ScreenOrientation.Orientation.PORTRAIT_DOWN
-            ) {
-                setPortrait();
+    useEffect(() => {
+        const onOrientationChange = (currentOrientation: ScreenOrientation.OrientationChangeEvent) => {
+            const orientationValue = currentOrientation.orientationInfo.orientation;
+            if (orientationValue == 1 || orientationValue == 2) {
                 setIsLandscape(false);
             } else {
-                setLandscape();
                 setIsLandscape(true);
             }
```

Comment:
I like this way of encoding the `landscape` hook. I just have a small refractor to propose using ternary operator to make this fit on one line.

```
setIsLandscape((orientationValue == 1 || orientationValue == 2) ? false : true)
``` 
I know you could squeeze it more, but I prefer to explicit the value so that we have no surprise with the `else` statement.

Reply:
Oh true this will make it cleaner indeed thanks